ggshield/utils/git_shell.py

```diff
@@ -234,6 +234,33 @@ def git_ls(wd: Optional[Union[str, Path]] = None) -> List[str]:
     return git(cmd, timeout=600, cwd=wd).split("\n")
 
 
+def _get_submodules_paths(wd: Optional[Union[str, Path]] = None) -> List[str]:
+    find_path = "git rev-parse --show-toplevel"
+    cmd = ["submodule", "--quiet", "foreach", find_path]
+    submodules = [
+        submodule
+        for submodule in git(cmd, timeout=600, cwd=wd).split("\n")
+        if submodule
+    ]
+
+    return submodules
```

Comment:
Also, using `.splitlines()` instead of `split("\n")` removes the need for removing trailing empty lines:

```
>>> a="fff\nbbb\n"
>>> a.split("\n")
['fff', 'bbb', '']
>>> a.splitlines()
['fff', 'bbb']
```